moveit_ros/visualization/motion_planning_rviz_plugin/src/motion_planning_frame_objects.cpp

```diff
@@ -268,7 +268,7 @@ void MotionPlanningFrame::selectedCollisionObjectChanged()
         else
           ui_->object_status->setText("ERROR: '" + sel[0]->text() + "' should be a collision object but it is not");
       }
-      if (update_scene_marker)
+      if (update_scene_marker && ui_->tabWidget->tabText(ui_->tabWidget->currentIndex()) == "Scene Objects")
```

Comment:
I stayed with the direct comparison because it was already in the code. But sure, we can refactor this too.
I added constants for all the titles and changed the required places.